cmd/init.go

```diff
@@ -0,0 +1,277 @@
+			return fmt.Errorf("failed to start Caddy: %w", err)
+		}
+
+		// wait 5s to make sure Caddy is up
+		time.Sleep(5 * time.Second)
+		err = caddyapi.InitCaddy()
+		if err != nil {
+			return fmt.Errorf("failed to init Caddy: %w", err)
+		}
+
+		fmt.Println("Caddy started successfully!")
+	}
+	
+	if o.certsDir != "" {
+		caddyCertsDir := caddyConfig + "/certs"
+		if _, err := os.Stat(caddyCertsDir); os.IsNotExist(err) {
+			if err := os.MkdirAll(caddyCertsDir, 0755); err != nil {
+				return fmt.Errorf("failed to create Caddy certs directory: %w", err)
+			}
+		}
+
+		certsDir := caddyCertsDir + "/" + o.domain
+		if _, err := os.Stat(certsDir); os.IsNotExist(err) {
+			if err := os.MkdirAll(certsDir, 0755); err != nil {
+				return fmt.Errorf("failed to create certs directory: %w", err)
+			}
+		}
+
+		certs, err := os.ReadDir(o.certsDir)
+		if err != nil {
+			return fmt.Errorf("failed to read certs directory: %w", err)
+		}
+
+		for _, cert := range certs {
+			if cert.IsDir() {
+				continue
+			}
+
+			certPath := o.certsDir + "/" + cert.Name()
+			newCertPath := certsDir + "/" + cert.Name()
+
+			bytes, err := os.ReadFile(certPath)
+			if err != nil {
+				return fmt.Errorf("failed to read cert file: %w", err)
+			}
+
+			if err := os.WriteFile(newCertPath, bytes, 0644); err != nil {
+				return fmt.Errorf("failed to copy cert file: %w", err)
+			}
+		}	
+
+		fmt.Println("Certs copied successfully!")
+	}
+	
+	// GitOps name
+	gitopsName := "gitops"
+	if len(args) == 1 {
+		gitopsName = args[0]
+	}
+
+	gitopsConfig := bitswanConfig + gitopsName
+	if _, err := os.Stat(gitopsConfig); !os.IsNotExist(err) {
+		return fmt.Errorf("GitOps with this name was already initialized: %s", gitopsName)
+	}
+
+	if err := os.MkdirAll(gitopsConfig, 0755); err != nil {
+		return fmt.Errorf("failed to create GitOps directory: %w", err)
+	}
+
+
+
+	// Initialize Bitswan workspace
+	gitopsWorkspace := gitopsConfig + "/workspace"
+	if o.remoteRepo != "" {
+		com := exec.Command("git", "clone", o.remoteRepo, gitopsWorkspace)
+
+		fmt.Println("Cloning remote repository...")
+		if err := com.Run(); err != nil {
+			return fmt.Errorf("Failed to clone remote repository: %w", err)
+		}
+		fmt.Println("Remote repository cloned!")
+	} else {
+
+		os.Mkdir(gitopsWorkspace, 0755)
+		com := exec.Command("git", "init")
+		com.Dir = gitopsWorkspace
+
+		fmt.Println("Initializing git in workspace...")
+		if err := com.Run(); err != nil {
+			return fmt.Errorf("Failed to init git in workspace: %w", err)
+		}
+		fmt.Println("Git initialized in workspace!")
+	}
+
+	// Add GitOps worktree
+	gitopsWorktree := gitopsConfig + "/gitops"
+	worktreeAddCom := exec.Command("git", "worktree", "add", "--orphan", "-b", gitopsName, gitopsWorktree)
+	worktreeAddCom.Dir = gitopsWorkspace
+
+	fmt.Println("Setting up GitOps worktree...")
+	if err := worktreeAddCom.Run(); err != nil {
+		return fmt.Errorf("failed to create GitOps worktree: %w", err)
+	}
+
+	if o.remoteRepo != "" {

```

Comment:
What if the remote repo is already initialized?

Reply:
The initial commit is needed to create the the branch on the origin so you can setup the push upstream..I haven't found any better way how to do this...you're right that this will mess already initialized repo...I think also `--orphan` worktree won't work...I think I added TODO somewhere that there should be check if the branch exists on the remote repo...what I was wondering is that if the branch for the gitops already exists and there is a bitswan.yaml file with pipelines and checksums...when you move the gitops to another server where the checksums are not presented how will it work?